core/src/main/java/feign/FeignException.java

```diff
@@ -40,6 +40,7 @@
   private static final long serialVersionUID = 0;
   private int status;
   private byte[] responseBody;
+  private Map<String, Collection<String>> responseHeaders;
   private Request request;
 
   protected FeignException(int status, String message, Throwable cause) {
```

Comment:
this exception is getting more and more complicated.... wondering if we should have one constructor that takes `Request` as argument and the other one that takes `Response` 

Anyway, just concerned that this class is getting more and more complicated.